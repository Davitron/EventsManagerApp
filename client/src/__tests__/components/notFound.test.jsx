import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import NotFound from '../../components/NotFound';

let mountedComponent;
let wrapper;

const getComponent = () => {
  if (!mountedComponent) {
    mountedComponent = shallow(<NotFound />);
  }
  return mountedComponent;
};

describe('NotFound Component Test', () => {
  beforeAll(() => {
    wrapper = getComponent();
  });

  it('component successfully rendered', () => {
    const divs = wrapper.find('div');
    expect(divs.length).toBe(2);
  });

  it('component successfully rendered', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
