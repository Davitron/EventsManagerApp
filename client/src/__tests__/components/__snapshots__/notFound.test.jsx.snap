// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Component Test component successfully rendered 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotFound />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="my-container"
        style={
          Object {
            "color": "black",
            "paddingTop": "20em",
          }
        }
      >
        <h1
          className="animated fadeInUp"
          style={
            Object {
              "fontSize": "50px",
            }
          }
        >
          Oops! Not Found
        </h1>
        <h3
          className="animated fadeInUp"
          style={
            Object {
              "fontSize": "20px",
              "paddingBottom": "60px",
            }
          }
        >
          Request is either invalid or the requested resouce does not exist.
        </h3>
      </div>,
      "className": "background",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="animated fadeInUp"
            style={
              Object {
                "fontSize": "50px",
              }
            }
          >
            Oops! Not Found
          </h1>,
          <h3
            className="animated fadeInUp"
            style={
              Object {
                "fontSize": "20px",
                "paddingBottom": "60px",
              }
            }
          >
            Request is either invalid or the requested resouce does not exist.
          </h3>,
        ],
        "className": "my-container",
        "style": Object {
          "color": "black",
          "paddingTop": "20em",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Oops! Not Found",
            "className": "animated fadeInUp",
            "style": Object {
              "fontSize": "50px",
            },
          },
          "ref": null,
          "rendered": "Oops! Not Found",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Request is either invalid or the requested resouce does not exist.",
            "className": "animated fadeInUp",
            "style": Object {
              "fontSize": "20px",
              "paddingBottom": "60px",
            },
          },
          "ref": null,
          "rendered": "Request is either invalid or the requested resouce does not exist.",
          "type": "h3",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="my-container"
          style={
            Object {
              "color": "black",
              "paddingTop": "20em",
            }
          }
        >
          <h1
            className="animated fadeInUp"
            style={
              Object {
                "fontSize": "50px",
              }
            }
          >
            Oops! Not Found
          </h1>
          <h3
            className="animated fadeInUp"
            style={
              Object {
                "fontSize": "20px",
                "paddingBottom": "60px",
              }
            }
          >
            Request is either invalid or the requested resouce does not exist.
          </h3>
        </div>,
        "className": "background",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="animated fadeInUp"
              style={
                Object {
                  "fontSize": "50px",
                }
              }
            >
              Oops! Not Found
            </h1>,
            <h3
              className="animated fadeInUp"
              style={
                Object {
                  "fontSize": "20px",
                  "paddingBottom": "60px",
                }
              }
            >
              Request is either invalid or the requested resouce does not exist.
            </h3>,
          ],
          "className": "my-container",
          "style": Object {
            "color": "black",
            "paddingTop": "20em",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Oops! Not Found",
              "className": "animated fadeInUp",
              "style": Object {
                "fontSize": "50px",
              },
            },
            "ref": null,
            "rendered": "Oops! Not Found",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Request is either invalid or the requested resouce does not exist.",
              "className": "animated fadeInUp",
              "style": Object {
                "fontSize": "20px",
                "paddingBottom": "60px",
              },
            },
            "ref": null,
            "rendered": "Request is either invalid or the requested resouce does not exist.",
            "type": "h3",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
